diff --git a/source/arm9/system/initSystem.c b/source/arm9/system/initSystem.c
index 3f6a5c8..aebddf1 100644
--- a/source/arm9/system/initSystem.c
+++ b/source/arm9/system/initSystem.c
@@ -47,6 +47,14 @@ extern time_t *punixTime;
 
 int __libnds_gtod(struct _reent *ptr, struct timeval *tp, struct timezone *tz);
 
+void __syscall_exit(int rc) {
+	__libnds_exit(rc);
+}
+
+int __syscall_gettod_r(struct _reent *ptr, struct timeval *tp, struct timezone *tz) {
+	return __libnds_gtod(ptr, tp, tz);
+}
+
 //---------------------------------------------------------------------------------
 // Reset the DS registers to sensible defaults
 //---------------------------------------------------------------------------------
@@ -93,9 +101,6 @@ void __attribute__((weak)) initSystem(void) {
 
 	punixTime = (time_t*)memUncached((void *)&__transferRegion()->unixTime);
 
-	__syscalls.exit = __libnds_exit;
-	__syscalls.gettod_r = __libnds_gtod;
-
 	extern  char *fake_heap_end;
 	__transferRegion()->bootcode = (struct __bootstub *)fake_heap_end;
 	irqEnable(IRQ_VBLANK);
